ambry-account/src/main/java/com/github/ambry/account/mysql/AccountDao.java

```diff
@@ -448,6 +453,53 @@ public synchronized Dataset getDataset(int accountId, int containerId, String ac
     }
   }
 
+  /**
+   * Get the latest version value of the dataset.
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param datasetName the name of the dataset.
+   * @return the latest version.
+   * @throws SQLException
+   */
+  public synchronized long getLatestVersion(short accountId, short containerId, String datasetName)
+      throws SQLException {
+    try {
+      long startTimeMs = System.currentTimeMillis();
+      dataAccessor.getDatabaseConnection(false);
+      PreparedStatement listVersionStatement = dataAccessor.getPreparedStatement(listVersionSql, false);
+      long latestVersion = executeListVersionStatement(listVersionStatement, accountId, containerId, datasetName);
+      dataAccessor.onSuccess(Read, System.currentTimeMillis() - startTimeMs);
+      return latestVersion;
+    } catch (SQLException e) {
+      dataAccessor.onException(e, Read);
+      throw e;
+    }
+  }
+
+  /**
+   * Execute the listVersionSql statement.
+   * @param statement the statement to list the latest version.
+   * @param accountId the id for the parent account.
+   * @param containerId the id of the container.
+   * @param datasetName the name of the dataset.
+   * @return the latest version of the dataset.
+   * @throws SQLException
+   */
+  private long executeListVersionStatement(PreparedStatement statement, int accountId, int containerId,
+      String datasetName) throws SQLException {
+    ResultSet resultSet = null;
+    try {
+      statement.setInt(1, accountId);
+      statement.setInt(2, containerId);
+      statement.setString(3, datasetName);
+      resultSet = statement.executeQuery();
```

Comment:
is there a way to specify timeouts for these queries ?

Reply:
I think we don't need to specify right now,  we have the mysql connection timeout, and if we hit the timeout, we can work with mysql team to tune it.